mvebu: Clean up dts file and others tweak

Ansuel opened the pull request:
Need testing for this... I removed some extra definition in rango dts file as they are present in the dtsi file
@mkresin 

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

mkresin commented:
Would you please leave a not as soon as this was tested!

anomeome commented:
She be borked for me.
  -did a git stash, and applied PRs 913,914,915(rango part), i.e. clean tree
  -compiled my usual build
  -flashed not keeping config
  -no detection of usb 1.x thumb on USB3(USB port2) device, no power
  -no detection of sata device on USB2/SATA(USB port1) device
  -restored my backed up config, no change

Note: change 915 DTS change is pre 913

Ansuel commented:
@anomeome need to know if all work correctly (not the sata led part)
This dts file is full of extra part that are included in the dtsi file (the general dts file for all linksys board) so i need someone to test this and tell me if he notice any difference or problem with this

(sata led is wip)

anomeome commented:
I'm not talking about LEDs. I'm saying things that worked no longer function. A USB device attached gets no power. The devices attached to the sata and USB3 port are no longer detected. That is as far as I took the testing, since they stood out seemed no point to got any further.

Ansuel commented:
@anomeome test this version pls
need to underestand how dts work... now all should work...

Ansuel commented:
I'm talking with a kernel guy to include the dts file upstream...
He says me that is wrong to include partition thing to the dts file and we need to separate that to another dtsi file... so here 
Need more testing for rango... (i tested this with shelby and is working right)

Ansuel commented:
shelby work good

nbd168 commented:
With that last commit, I don't really understand how this is making things better. For instance, moving things to a .dtsi file that is included only once doesn't seem like a good idea to me.

Ansuel commented:
Ask that to the kernel guy ahahahha 
They only accept led definition in the dts file
Switch definition and partition need to be in a dtsi file

I with your opinion that create another dtsi file just for rango is useless... Anyway if you see all in another way, if linksys relase another device with similart specs to rango we can support it more quickly

nbd168 commented:
Do you have a link to the email thread?

Ansuel commented:
i just paste here the last message

(i think he don't know the difference between rango and other, but he mean that we need a more ramificated version with the dts)
This commit should be the way he describe

For the GPIO LEDs look at cobra, caiman as examples. The LEDs for
rango go in rango.dts.

The flash@d0000 node in armada-385-linksys.dtsi should be moved into
another .dtsi file, e.g. armada-385-linksys-caiman-cobra.dtsi.

So you end up with

armada-385-linksys-caiman.dts
 - armada-385-linksys.dtsi
 - armada-385-linksys-caiman-cobra.dtsi

armada-385-linksys-cobra.dts
 - armada-385-linksys.dtsi
 - armada-385-linksys-caiman-cobra.dtsi

armada-385-linksys-rango.dts
 - armada-385-linksys.dtsi

Ansuel commented:
thx to @anomeome that report me problem with patch and dts file
I included them to the dts file so we can drop it

anomeome commented:
Functionality tested fine on mamba and rango.

mkresin commented:
I really don't want to be the killjoy here, but it doesn't work this way.

We do not overwrite kernel files with our own versions. We patch them. Same applies to backports from more recent kernel version or your own pending patches. We add them as patches.

If you are getting in conflict with the LEDE dts files, rename the files folder to files-4.4 to make sure it is only used for kernel 4.4.

@Ansuel would you please merge your three PRs into one. It would make reviewing for me way easier. It would be the best to add the missing commits to this PR.

mkresin commented:
@Ansuel would you please paste a link to the upstream discussion here so that we can see what kind of changes are requested.

Ansuel commented:
@mkresin bad day today (lots of test..) anyway, i paste here the one that talk about the modification to do...
I modify the 3 commit...
One is just the dts file (original) with the trigger added...
one is just the simol to add to the kernel anyway 
This one is the pr with lots of changes 
Can't merge all of them intro one cause they do different thing
Anyway i get a message form the maintainer of the linksys board @kaloz about the dts file patch (shelby and rango support in kernel)

> NAK. Patch to support shelby has been sent upstream by Ralph - the rango dts isn't finalized yet, hence it hasn't been sent upstream.
> 
> 
> Best,
> 
> Imre

Wouldn't be useless to specificy the dts file for kernel-4.9 only? i mean... we moved to to kernel 4.9 with all linksys board, right?

thagabe commented:
So mamba, cobra, caiman have been upstream? Isn't there only 2 DTS files 1 for cobra and 2) rango? The changes for mamba and caiman are just patches applied on top of the cobra one? Correct me if I'm wrong because I really want to learn more. Also could @Ansuel link to the "upstream" chat? When referring to upstream do you mean LEDE upstream (their git repo) or kernel upstream? THXs

Ansuel commented:
@thagabe From what I understand, in the kernel we have only dts for mamba cobra and caiman... Its lack of Shelby and rango Shelby is pushed upstream, rango is missing something (this is what the real maintainers told me with an email) upstream = kernel not lede
This mess started when I tried to push the 2 dts file that we use "locally" for Shelby and rango. A kernel guy told me that in the dts file should be only led related stuff, partition, switch driver and all other specific things should be in a dtsi file...
So with this commit I create a more specific dts file 
shelby caiman and cobra share the same partition table and the same switch so a create a dtsi file that is only specific for this 3 board, then i modified the linksys dtsi file so that it contains only shared code between rango shelby caiman and cobra. Then i created the rango dtsi and i removed from the rango dts all the stuff except the led part. I needed someone to test these changes as i only have a shelby router but not a rango one... @anomeome help me with testing the rango changes so now i can push this to the kernel and see if they complain something about. 
We can also include this to lede as we can drop 2-3 patch with this changes.
Is it clear now?

thagabe commented:
Indeed it is. Thanks @Ansuel

hnyman commented on target/linux/mvebu/files/arch/arm/boot/dts/armada-385-linksys-rango.dtsi:
```diff
@@ -49,12 +49,15 @@
 	
 	
 	soc {
+<<<<<<< HEAD
```
Merge garbage still left here.
Looks untested to me...

Ansuel commented:
@hnyman it's tested by me and others with rango and is working good
that was a mistake by me when i resolved some conflicts with the recent changes

Ansuel commented:
Will try to push this upstream, thx for testing.
